<a href="https://colab.research.google.com/github/phantrang184/Data-analysis-in-business-lecture-assignment/blob/main/1_Th%E1%BB%91ng_k%C3%AA_m%C3%B4_t%E1%BA%A3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import statsmodels.api as sm

In [ ]:
df = pd.read_csv('/content/drive/MyDrive/Train.csv',
                 index_col=0)

In [ ]:
df


,Warehouse_block,Mode_of_Shipment,Customer_care_calls,Customer_rating,Cost_of_the_Product,Prior_purchases,Product_importance,Gender,Discount_offered,Weight_in_gms,Reached.on.Time_Y.N
ID,,,,,,,,,,,
1,D,Flight,4,2,177,3,low,F,44,1233,1
2,F,Flight,4,5,216,2,low,M,59,3088,1
3,A,Flight,2,2,183,4,low,M,48,3374,1
4,B,Flight,3,3,176,4,medium,M,10,1177,1
5,C,Flight,2,2,184,3,medium,F,46,2484,1
...,...,...,...,...,...,...,...,...,...,...,...
10995,A,Ship,4,1,252,5,medium,F,1,1538,1
10996,B,Ship,4,1,232,5,medium,F,6,1247,0
10997,C,Ship,5,4,242,5,low,F,4,1155,0


In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 10999 entries, 1 to 10999
Data columns (total 11 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   Warehouse_block      10999 non-null  object
 1   Mode_of_Shipment     10999 non-null  object
 2   Customer_care_calls  10999 non-null  int64 
 3   Customer_rating      10999 non-null  int64 
 4   Cost_of_the_Product  10999 non-null  int64 
 5   Prior_purchases      10999 non-null  int64 
 6   Product_importance   10999 non-null  object
 7   Gender               10999 non-null  object
 8   Discount_offered     10999 non-null  int64 
 9   Weight_in_gms        10999 non-null  int64 
 10  Reached.on.Time_Y.N  10999 non-null  int64 
dtypes: int64(7), object(4)
memory usage: 1.0+ MB


In [ ]:
#thống kê mô tả cho dữ liệu
(df.describe(include="all"))

,Warehouse_block,Mode_of_Shipment,Customer_care_calls,Customer_rating,Cost_of_the_Product,Prior_purchases,Product_importance,Gender,Discount_offered,Weight_in_gms,Reached.on.Time_Y.N
count,10999,10999,10999.000000,10999.000000,10999.000000,10999.000000,10999,10999,10999.000000,10999.000000,10999.000000
unique,5,3,NaN,NaN,NaN,NaN,3,2,NaN,NaN,NaN
top,F,Ship,NaN,NaN,NaN,NaN,low,F,NaN,NaN,NaN
freq,3666,7462,NaN,NaN,NaN,NaN,5297,5545,NaN,NaN,NaN
mean,NaN,NaN,4.054459,2.990545,210.196836,3.567597,NaN,NaN,13.373216,3634.016729,0.596691
std,NaN,NaN,1.141490,1.413603,48.063272,1.522860,NaN,NaN,16.205527,1635.377251,0.490584
min,NaN,NaN,2.000000,1.000000,96.000000,2.000000,NaN,NaN,1.000000,1001.000000,0.000000
25%,NaN,NaN,3.000000,2.000000,169.000000,3.000000,NaN,NaN,4.000000,1839.500000,0.000000
50%,NaN,NaN,4.000000,3.000000,214.000000,3.000000,NaN,NaN,7.000000,4149.000000,1.000000
75%,NaN,NaN,5.000000,4.000000,251.000000,4.000000,NaN,NaN,10.000000,5050.000000,1.000000


Các biến số
Customer care calls: Trung bình nhận được 4.05 cuộc gọi mỗi đơn, độ lệch chuẩn khoảng 1.14 cuộc gọi cho thấy khá ổn định. Các khoảng tứ phân vị cho thấy lượng cuộc gọi được phân bổ tăng khá đều và giảm khá đều ở mỗi 25% lượng khách. Trong đó nhiều nhất một khách sẽ gọi 7 lần và ít nhất một khách gọi 2 lần.


Customer rating: Đánh giá trung bình của khách đang ở mức 3 / 5 -> Tương đối thấp. Và số tỉ lê phân bố mức đánh giá ở từng khoảng tứ phân vị khá đồng đều có thể dự đoán được số lượng đánh giá từ 1-5 sao đều đều nhau. Điều này chứng tỏ sản phẩm chưa đáp ứng sự hài lòng của khách hàng, và có thể chất lượng sản phẩm/ dịch vụ không đồng đều.

Cost of the product: Giá tiền của sản phẩm dao động từ $96 đến $310, trung bình mỗi sản phẩm có giá $210. Có thể thấy giá tiền mỗi đơn hàng được đặt là tương đối cao.

Prior purchases: Số lần mua hàng trước đó trung bình là 3.56 lần. Tối thiểu một khách hàng đã mua trước đó 2 lần và nhiều nhất là 10 lần. Điều này cho thấy tất cả đều là khách hàng quen thuộc, không có khách hàng mới tham gia ecommerce. Tuy nhiên cũng có thể nhận thấy có một vài trường hợp đặc biệt có khách hàng mua rất nhiều lên đến 10 lần, có thể xem đây là nhóm khách hàng vip cần quan tâm đặc biệt.

Discount offered: Mức giảm giá trung bình là $13.37, với độ lệch chuẩn rất lớn lên đến $16.21, cho thấy mức giảm giá có sự phân tán rất cao có thể phụ thuộc vào giá trị món hàng và số lần mua hàng của khách.

Weight in gms: Trung bình một món hàng nặng 3634 grams, món nặng nhất lên đến 7846 grams, chứng tỏ hàng hóa vận chuyển đều là hàng nặng, với độ lệch chuẩn lên đến 1635 grams còn cho thấy hàng hóa có độ đa dạng về cân nặng lớn. Có thể kết luận công ty ecommerce trên chỉ vận chuyển các mặt hàng lớn, có thể cồng kềnh, hoặc hàng sỉ theo lô.


Các biến phân loại:
Warehouse block: Có đến 5 kho hàng A B C D F, trong đó kho hàng F sử dụng nhiều nhất lên đến 3666 lần.

Mode of shipment: Vận chuyển gồm 3 hình thức và Ship là hình thức được sử dụng nhiều nhất tới 7462 lần.

Product importance: Phần lớn hàng đều là hàng ở mức low importance, ít hàng high và medium importance

Gender: Phân bổ tỉ lệ khách nam nữ có thể khá bằng nhau nhưng nữ chiếm nhiều hơn với 5545 khách là nữ.

Các biến mục tiêu
Reached on time YN: Tỷ lệ trung bình giao hàng đúng giờ chỉ là 36.34% với độ lệch chuẩn 16.35% (khá thấp)
